### PR TITLE
WIP: static url

### PR DESCRIPTION
**Pull request by lunny**
This PR will allow static resources served by another domain different from the main domain. So that a CDN service could be used easier.

### CODE REVIEW

**Comment by adelowo** [comment]
We should add it to the docs too not just `ini.sample`

File: modules/templates/helper.go [hunk 1]
```diff
@@ -108,12 +111,12 @@ func NewFuncMap() []template.FuncMap {
...
-		"DiffLineTypeToStr":     DiffLineTypeToStr,
-		"Sha1":                  Sha1,
-		"ShortSha":              base.ShortSha,
-		"MD5":                   base.EncodeMD5,
+		"EllipsisString":    base.EllipsisString,
+		"DiffTypeToStr":     DiffTypeToStr,
+		"DiffLineTypeToStr": DiffLineTypeToStr,
+		"Sha1":              Sha1,
+		"ShortSha":          base.ShortSha,
+		"MD5":               base.EncodeMD5,
```
Comment (das7pad) [inline modules/templates/helper.go:117]: Please run go fmt

Diff: https://github.com/das7pad/gitea/commit/f219cacd96c1d2ae3e9a393c2651a17ada48df0d

**Comment by das7pad** [comment]
Hi @lunny, I continued your work over at this branch [lunny/static_url in my fork](https://github.com/das7pad/gitea/commits/lunny/static_url).

All resources from `/public` should be converted by now, as in replacing `{{AppSubUrl}}/PATH` -> `{{StaticUrlPrefix}}/PATH` with `PATH` being css, img, js, less (no hit here), vendor.

**Comment by lunny** [comment]
Please send a new PR to gitea.

**Comment by das7pad** [comment]
PR landed in #7911

**Pull request closed without merging**